runtime/lua/vim/clipboard/osc52.lua

```diff
@@ -20,19 +20,42 @@ function M.paste()
 
   io.stdout:write('\x1b]52;;?\x1b\\')
 
-  vim.wait(1000, function()
+  local ok, res
+
+  -- Wait 1s first for terminals that respond quickly
+  ok, res = vim.wait(1000, function()
     return contents ~= nil
   end)
 
-  -- Delete the autocommand if it didn't already delete itself
-  pcall(vim.api.nvim_del_autocmd, id)
+  if res == -1 then
+    -- If no response was received after 1s, print a message and keep waiting
+    vim.api.nvim_echo(
+      { { 'Waiting for OSC 52 response from the terminal. Press Ctrl-C to interrupt...' } },
+      false,
+      {}
+    )
+    vim.cmd.redraw()
```

Comment:
If https://github.com/neovim/neovim/pull/25937 is merged:

```suggestion
      { flush = true }
    )
```